docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,786 @@
+
+### Built in JSON support :thumbsup:
+
+pydantic-core can parse JSON directly into a model or output type, this both improves performance and avoids
+issue with strictness - e.g. if you have a strict model with a `datetime` field, the input must be a 
+`datetime` object, but clearly that makes no sense when parsing JSON which has no `datatime` type.
+Same with `bytes` and many other types.
+
+Pydantic V2 will therefore allow some conversion when validating JSON directly, even in strict mode 
+(e.g. `ISO8601 string -> datetime`, `str -> bytes`) even though this would not be allowed when validating
+a python object.
+
+In future direct validation of JSON will also allow:
+
+* parsing in a separate thread while starting validation in the main thread
+* line numbers from JSON to be included in the validation errors
+
+!!! note
+    Pydantic has always had special support for JSON, that is not going to change.
+
+    While in theory other formats
+    could be specifically supported, the overheads are significant and I don't think there's another format that's
+    used widely enough to be worth specific logic. Other formats can be parsed to python then validated, similarly
+    when serialising, data can be exported to a python object, then serialised, 
+    see [below](#improvements-to-dumpingserializationexport).
+
+### Validation without a Model :thumbsup:
+
+In pydantic V1 the core of all validation was a pydantic model, this led to significant overheads and complexity
+when the output data type was not a model.
+
+pydantic-core operates on a tree of validators with no "model" type required at the base of that tree.
+It can therefore validate a single `string` or `datetime` value, a `TypeDict` or a `Model` equally easily.
+
+This feature will provide significant addition performance improvements in scenarios like:
+
+* Adding validation to `dataclasses`
+* Validating URL arguments, query strings, headers, etc. in FastAPI
+* Adding validation to `TypedDict`
+* Function argument validation
+* Adding validation to your custom classes, decorators...
+
+In effect - anywhere where you don't care about a traditional model class instance.
+
+We'll need to add standalone methods for generating JSON Schema and dumping these objects to JSON, etc.
+
+### Required vs. Nullable Cleanup :thumbsup:
+
+Pydantic previously had a somewhat confused idea about "required" vs. "nullable". This mostly resulted from
+my misgivings about marking a field as `Optional[int]` but requiring a value to be provided but allowing it to be 
+`None`.
+
+In pydantic V2, pydantic will move to match dataclasses, thus:
+
+```py title="Required vs. Nullable"
+from pydantic import BaseModel
+
+class Foo(BaseModel):
+    f1: str  # required, cannot be None
+    f2: str | None  # required, can be None - same as Optional[str] / Union[str, None]
+    f3: str | None = None  # optional, can be None
+    f4: str = 'Foobar'  # optional, but cannot be None  
```

Comment:
```suggestion
    f3: str | None = None  # not required, can be None
    f4: str = 'Foobar'  # not required, but cannot be None  
```

I'd rephrase this for extra clarity 🙂  (and I'm super excited about the cleanup!)